src/disco/events/fd_circbuf.c

```diff
@@ -0,0 +1,160 @@
+#include "fd_circbuf.h"
+
+struct __attribute__((aligned(8UL))) fd_circbuf_message_private {
+  ulong align;
+  ulong footprint;
+  ulong next;
```

Comment:
Does the next field store unique information? Is it just current+footprint with the proper alignment?